concepts.wip/if-expressions/about.md

```diff
@@ -6,10 +6,30 @@ In Julia, like many other languages, allow for the following with `if`-expressio
 - Nested if-expressions
 
 ~~~~exercism/note
-`if`-expressions as defined here will encompass the following syntaxes:`if`, `if-else` and `if-elseif-else` statements.
+[`if`-expressions](https://docs.julialang.org/en/v1/manual/control-flow/#man-conditional-evaluation) as defined here will encompass the following syntaxes:`if`, `if-else` and `if-elseif-else` statements.
 ~~~~
 
-`if`-expressions allows for branching in our programs, given a conditional is satisfied. 
+`if`-expressions allows for branching in our programs, given a condition is satisfied.
+
+## A prompt program example
+
+Suppose you've defined a `password_checker(password)` function that returns `true`, if the password entered by the user is correct.
+
+```julia
+# Login prompt
+println("Enter Password: ")
+
+# Get user input
+password = readline()
+
+# Our main logic
+if password_checker(password)
+    println("Welcome!")
+end
+```
+Although the above program is not very realistic, but it shows the anatomy of an `if` statement. 
+
+In this case the `println("Welcome!")` is run, if the `password_checker(password)` returns `true`, otherwise it is skipped.
```

Comment:
This is mostly a personal view, not an Exercism view, even though some other maintainers agree:

Imho any examples with reading input or printing anything should be avoided completely. So many people new to programming struggle with understanding the difference between printing and returning a value, often because their introductory classes or texts use printing everywhere. This is even more confusing in REPL settings where the difference is visually indistinguishable. To avoid this, I'd only print in an exercise that's explicitly about IO, not general examples or other exercises.

The example itself works but perhaps it could be turned into a function `password_check(password)`?

Reply:
Thank you for explaining this. Understood, I've converted the example into a `function`, as suggested. Please let me know if the example is clear.

Reply:
That's clearer, though I'd change it to return a string